src/windows.rs

```diff
@@ -107,3 +110,172 @@ impl PutAddress for ProcessHandle {
         }
     }
 }
+
+/// Use `CreateToolhelp32Snapshot` to get and filter list of loaded modules (called DLLs in Windows)
+/// of this process, returning the base address of it.
+#[allow(clippy::clippy::cast_possible_truncation)] // for size_of as u32
+impl ModuleInfo for Pid {
+    fn get_module_base(&self, name: &str) -> std::io::Result<usize> {
+        // taken from https://stackoverflow.com/questions/41552466/how-do-i-get-the-physical-baseaddress-of-an-dll-used-in-a-process
+        let mut module_entry = tlhelp32::MODULEENTRY32 {
+            dwSize: 0,
+            th32ModuleID: 0,
+            th32ProcessID: 0,
+            GlblcntUsage: 0,
+            ProccntUsage: 0,
+            modBaseAddr: std::ptr::null_mut(), // yikes
+            modBaseSize: 0,
+            hModule: std::ptr::null_mut(), // yikes
+            szModule: [0; tlhelp32::MAX_MODULE_NAME32 + 1],
+            szExePath: [0; winapi::shared::minwindef::MAX_PATH],
+        };
+
+        unsafe {
+            module_entry.dwSize = std::mem::size_of::<tlhelp32::MODULEENTRY32>() as u32;
+
+            let snapshot = tlhelp32::CreateToolhelp32Snapshot(
+                tlhelp32::TH32CS_SNAPMODULE | tlhelp32::TH32CS_SNAPMODULE32,
+                *self,
+            );
+            if snapshot == INVALID_HANDLE_VALUE {
+                return Err(std::io::Error::last_os_error());
+            }
+
+            if tlhelp32::Module32First(snapshot, &mut module_entry) == minwindef::TRUE {
```

Comment:
Why not use:
```rs
while tlhelp32::Module32Next(snapshot, &mut module_entry) == minwindef::TRUE {
    ...
}
```

What does `Module32Next` do when you give it a null module entry?

Reply:
Because we don't want to ignore the first entry returned by `Module32First`. (Same goes for `Process32First`), and rust has no do-while loops, which is what I assume the windows devs designed those functions around?